File: LmsApplication.CourseBoardModule.Data/Models/PostModel.cs
using LmsApplication.Core.Shared.Models;
using LmsApplication.CourseBoardModule.Data.Entities;

namespace LmsApplication.CourseBoardModule.Data.Models;

public class PostModel
{
    public required Guid Id { get; set; }
    
    public required string Content { get; set; }
    
    public required Guid EditionId { get; set; }
    
    public required UserExchangeModel Author { get; set; }
    
    public required ReactionModel Reactions { get; set; }
    
    public required ReactionType? CurrentUserReaction { get; set; }
    
    public required int CommentsCount { get; set; }
    
    public required DateTime CreatedAt { get; set; }
    
    public required DateTime? UpdatedAt { get; set; }
}

File: LmsApplication/LmsApplication.Frontend/src/app/types/course-board/post-model.ts
import { UserModel } from '../users/user-model';
import { ReactionModel } from './reaction-model';
import { ReactionType } from './reaction-type';

export interface PostModel {
  id: string;
  content: string;
  editionId: string;
  author: UserModel;
  reactions: ReactionModel;
  currentUserReaction: ReactionType | null;
  commentsCount: number;
  createdAt: Date;
  updatedAt: Date;
}
